compileopts/config.go

```diff
@@ -265,11 +265,14 @@ func (c *Config) CFlags(libclang bool) []string {
 	}
 	resourceDir := goenv.ClangResourceDir(libclang)
 	if resourceDir != "" {
-		// The resoure directory contains the built-in clang headers like
+		// The resource directory contains the built-in clang headers like
 		// stdbool.h, stdint.h, float.h, etc.
 		// It is left empty if we're using an external compiler (that already
 		// knows these headers).
-		cflags = append(cflags, "-resource-dir="+resourceDir)
+		cflags = append(cflags,
+			"-resource-dir="+resourceDir,
+			"-isystem", filepath.Join(resourceDir, "include"), // needed by Xtensa
+		)
```

Comment:
Can you do this in a way that only applies to Xtensa?

Something like:

```go
if strings.HasPrefix(c.Triple(), "xtensa") {
    // workaround needed in LLVM 16, see: https://github.com/espressif/llvm-project/issues/83
    cflags = append(cflags, "-isystem", filepath.Join(resourceDir, "include"))
}
```

This makes it clear that it is a workaround that can hopefully be removed in the future. And makes sure it only applies to Xtensa, to avoid affecting other platforms.

Also, I suspect a similar change will be needed in builder/library.go (test after `tinygo clean` to be sure).

Reply:
Updated `configopts/config.go` based on your suggestion.

I did not need any change to `builder/library.go` even after a `tinygo clean`